src/ip.cpp

```diff
@@ -167,7 +167,11 @@ int zmq::get_peer_ip_address (fd_t sockfd_, std::string &ip_addr_)
     }
 #else
     if (rc == -1) {
+#if !defined(TARGET_IPHONE) || !TARGET_IPHONE
```

Comment:
Where does TARGET_IPHONE come from?

Reply:
I believed it to be defined by the toolchain but that may not be true.  It's rather difficult to get the cross compiling on OSX to iOS to work and somebody else wrote the script that sets up the environment to enable it.  Looks like I can rely on the compiler defining __APPLE__ and __MACH__ but I should conditionally on them #include <TargetConditionals.h> to ensure we get TARGET_IPHONE.  Will do.

Reply:
If you manage to set up a build script, could you send a PR that adds a travis-ci job building for iOS?